packages/react-native/ReactCommon/react/renderer/graphics/platform/ios/react/renderer/graphics/HostPlatformColor.mm

```diff
@@ -0,0 +1,115 @@
+{
+  int32_t light = dynamicColor.lightColor;
+  int32_t dark = dynamicColor.darkColor;
+  int32_t highContrastLight = dynamicColor.highContrastLightColor;
+  int32_t highContrastDark = dynamicColor.highContrastDarkColor;
+
+  UIColor *lightColor = [RCTConvert UIColor:@(light)];
+  UIColor *darkColor = [RCTConvert UIColor:@(dark)];
+  UIColor *highContrastLightColor = [RCTConvert UIColor:@(highContrastLight)];
+  UIColor *highContrastDarkColor = [RCTConvert UIColor:@(highContrastDark)];
+
+  if (lightColor != nil && darkColor != nil) {
+    UIColor *color = [UIColor colorWithDynamicProvider:^UIColor *_Nonnull(UITraitCollection *_Nonnull collection) {
+      if (collection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastDarkColor != nil) {
+          return highContrastDarkColor;
+        } else {
+          return darkColor;
+        }
+      } else {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastLightColor != nil) {
+          return highContrastLightColor;
+        } else {
+          return lightColor;
+        }
+      }
+    }];
+    return color;
+  } else {
+    return nil;
+  }
+
+  return nil;
+}
+
+static inline int32_t _ColorFromUIColor(UIColor *color)
+{
+  float ratio = 255;
+  CGFloat rgba[4];
+  [color getRed:&rgba[0] green:&rgba[1] blue:&rgba[2] alpha:&rgba[3]];
+  return ((int32_t)round((float)rgba[3] * ratio) & 0xff) << 24 | ((int)round((float)rgba[0] * ratio) & 0xff) << 16 |
+      ((int)round((float)rgba[1] * ratio) & 0xff) << 8 | ((int)round((float)rgba[2] * ratio) & 0xff);
+}
+
+int32_t RCTPlatformColorGetColorFromUIColor(const std::shared_ptr<void> uiColor)
+{
+  UIColor *color = (UIColor *)unwrapManagedObject(uiColor);
+  if (color) {
+    UITraitCollection *currentTraitCollection = [UITraitCollection currentTraitCollection];
+    color = [color resolvedColorWithTraitCollection:currentTraitCollection];
+    return _ColorFromUIColor(color);
+  }
+
+  return 0;
+}
+
+inline UIColor *_Nullable RCTUIColorFromComponentsColor(const facebook::react::ColorComponents &components)
+{
+  return [UIColor colorWithRed:components.red green:components.green blue:components.blue alpha:components.alpha];
+}
+
+Color::Color(int32_t color)
+{
+  uiColor_ = wrapManagedObject([RCTConvert UIColor:@(color)]);
+}
+
+Color::Color(DynamicColor dynamicColor)
+{
+  uiColor_ = wrapManagedObject(RCTUIColorFromDynamicColor(dynamicColor));
+}
+
+Color::Color(ColorComponents components)
+{
+  uiColor_ = wrapManagedObject(RCTUIColorFromComponentsColor(components));
+}
+
+bool Color::operator==(const Color &other) const
+{
+  return [unwrapManagedObject(getUIColor()) isEqual:unwrapManagedObject(other.getUIColor())];
+}
+
+bool Color::operator!=(const Color &other) const
+{
+  return ![unwrapManagedObject(getUIColor()) isEqual:unwrapManagedObject(other.getUIColor())];
```

Comment:
I think you can use the `==` operator from above. Something like `!(this == other)`

Reply:
@cipolleschi Done~